plugins/parsers/hep/parser.go

```diff
@@ -0,0 +1,226 @@
+package hep
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+var headerNames = map[string]int{"version": 1,
+	"protocol":  2,
+	"src_ip4":   3,
+	"dest_ip4":  4,
+	"src_ip6":   5,
+	"dest_ip6":  6,
+	"src_port":  7,
+	"dest_port": 8,
+	"tsec":      9,
+	//"tmsec":      10,
+	"proto_type": 11,
+	"node_id":    12,
+	"mode_pw":    14,
+	"cid":        17,
+	"vlan":       18,
+	"nodename":   19,
+}
+
+var headerReverseMap = map[int]string{1: "version",
+	2: "protocol",
+	3: "src_ip4",
+	4: "dest_ip4",
+	5: "src_ip6",
+	6: "dest_ip6",
+	7: "src_port",
+	8: "dest_port",
+	9: "tsec",
+	//10: "tmsec",
+	11: "proto_type",
+	12: "node_id",
+	14: "node_pw",
+	17: "cid",
+	18: "vlan",
+	19: "nodename",
+}
+
+type Parser struct {
+	//Json fields
+	MetricName         string
+	HepMeasurementName string
+	TagKeys            []string
+	JSONNameKey        string
+	JSONStringFields   []string
+	JSONQuery          string
+	JSONTimeKey        string
+	JSONTimeFormat     string
+	JSONTimezone       string
```

Comment:
You should embedd the [JSON parser config struct](https://github.com/influxdata/telegraf/blob/43017559fa77149a432f1e842ae9eab9db9b7441/plugins/parsers/json/parser.go#L24) here instead of redefining the entries.